sound/soc/sof/sof-client.h

```diff
@@ -0,0 +1,62 @@
+/* SPDX-License-Identifier: (GPL-2.0 OR BSD-3-Clause) */
+/*
+ * This file is provided under a dual BSD/GPLv2 license.  When using or
+ * redistributing this file, you may do so under either license.
+ *
+ * Copyright(c) 2020 Intel Corporation. All rights reserved.
+ *
+ * Author: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>
+ */
+
+#ifndef __SOUND_SOC_SOF_CLIENT_H
+#define __SOUND_SOC_SOF_CLIENT_H
+
+#include <linux/virtual_bus.h>
+
+enum sof_client_type {
+	SOF_CLIENT_AUDIO,
+	SOF_CLIENT_IPC,
+};
+
+/* SOF client device */
+struct sof_client_dev {
+	struct virtbus_device vdev;
+	void *data;
+};
+
+struct sof_client_drv {
+	const char *name;
+	enum sof_client_type type;
+	struct sof_client_ops *ops;
+	struct virtbus_driver *virtbus_drv;
+	struct list_head list;	/* item in SOF core client drv list */
+};
+
+/* client-specific ops, all optional */
+struct sof_client_ops {
+	/* check if client is requesting DSP to remain in D0 during suspend */
+	bool (*pm_remain_in_D0)(struct device *dev);
```

Comment:
what would the argument point to? the parent device?

Reply:
No, I intended this to point to the virtbus_device's dev

Reply:
let me change to sof_client_dev to be more explicit.